picard/file.py

```diff
@@ -305,9 +308,14 @@ def _copy_file_info_tags(self, to_metadata, from_metadata):
             to_metadata[info] = from_metadata[info]
 
     def copy_metadata(self, metadata, preserve_deleted=True):
-        acoustid = self.metadata["acoustid_id"]
         saved_metadata = {}
 
+        # Keep current value for special tags that got calculated from audio content
+        for tag in CALCULATED_TAGS:
+            if tag not in metadata.deleted_tags and self.metadata[tag]:
+                saved_metadata[tag] = self.metadata[tag]
+
```

Comment:
I wonder if this should not be after next loop instead

Reply:
Actually that looks correct as is

Reply:
> Actually that looks correct as is

@zas Looking at it again there is actually a functional difference in the case you put one of the tags, let's say `acoustid_id`  into preserved tags also.

**Previous behavior:**

1. The current value of `acoustid_id` from `file.metadata` gets saved
2. Due to preserved tags the value of `acoustid_id` from `file.orig_metadata`  gets saved
3. Metadata update gets applied
4. `acoustid_id`  value from `file.orig_metadata` gets restored
5. `acoustid_id`  value from `file.metadata` gets restored

Result: The value from `file.metadata` is kept

**New behavior:**

1. The current value of `acoustid_id` from `file.metadata` gets saved
2. Due to preserved tags the value of `acoustid_id` from `file.orig_metadata`  gets saved, overwriting the previously saved value
3. Metadata update gets applied
4. Saved `acoustid_id`  value `file.orig_metadata` gets restored

Result: The value from `file.orig_metadata` is restored

It is really a special case and only applies if the tag has a original value and is also put into the preserved tags configuration.

I'm actually not sure what is expected here. The old behavior kind of disabled the kept tags configuration. So maybe the new behavior is closer to expectation?

Reply:
@phw yes, I think new behavior is better, so let's keep it as is.